src/multivariate/mvnormalcanon.jl

```diff
@@ -118,3 +118,5 @@ unwhiten_winv!(J::ScalMat, x::AbstractVecOrMat) = whiten!(J, x)
 
 _rand!(rng::AbstractRNG, d::MvNormalCanon, x::AbstractMatrix) = add!(unwhiten_winv!(d.J, randn!(rng,x)), d.μ)
 _rand!(d::MvNormalCanon, x::AbstractMatrix) = _rand!(Base.GLOBAL_RNG, d, x)
+_rand!(rng::AbstractRNG, d::MvNormalCanon, x::AbstractVector) = add!(unwhiten_winv!(d.J, randn!(rng,x)), d.μ)
```

Comment:
Apparently not - AbstractVecOrMat is less specific and then the method is ambiguous, I made the same mistake when copying @getzdan s two lines,  https://github.com/JuliaStats/Distributions.jl/pull/625/commits/cee75c3692d924471cade1168063ea3426641582